library/alloc/src/collections/linked_list/tests.rs

```diff
@@ -1030,16 +1031,14 @@ fn extract_if_drop_panic_leak() {
 #[test]
 #[cfg_attr(not(panic = "unwind"), ignore = "test requires unwinding support")]
 fn extract_if_pred_panic_leak() {
-    static mut DROPS: i32 = 0;
+    thread_local! {static DROPS: Cell<i32> = Cell::new(0);}
 
     #[derive(Debug)]
     struct D(u32);
 
     impl Drop for D {
         fn drop(&mut self) {
-            unsafe {
-                DROPS += 1;
-            }
+            DROPS.with(|drops| drops.update(|v| v + 1));
```

Comment:
Could be written as `DROPS.set(DROPS.get() + 1)` thanks to the convenience methods in `LocalKey<Cell<T>>`.  Debatable if that’s better in this case but definitely nicer for cases where you only read or only write.

Reply:
I think it's already nicer even in this case, thanks!

And you think the thread-local is appropriate here?

Reply:
Sorry, I don't have enough context to say whether the overall approach is right, I just had a drive-by comment about the way it's implemented.

Reply:
Thanks, anyway!